server/location/locationController.js

```diff
@@ -0,0 +1,35 @@
+var distance = require('google-distance-matrix');
+var googlekeys = require('../config/googleconfig.js');
+
+distance.key(googlekeys.distanceMatrixKey);
+distance.units('imperial');
+
+module.exports = {
+
+  getDistance: function (request, response) {
+    distance.mode(request.body.transport.toLowerCase());
+
+    var arrOrigin = [request.body.origin];
+    var arrRestaurant = [request.body.restaurant];
+
+    distance.matrix(arrOrigin, arrRestaurant, function (error, distances) {
```

Comment:
Can you specify what distances is in the callback, and what the different properties (rows, status, etc.) refer to?